recipes/icu/all/conanfile.py

```diff
@@ -119,22 +132,6 @@ def generate(self):
         if cross_building(self):
             base_path = unix_path(self, self.dependencies.build["icu"].package_folder)
             tc.configure_args.append(f"--with-cross-build={base_path}")
-            if (not is_msvc(self)):
-                # --with-cross-build above prevents tc.generate() from setting --build option.
-                # Workaround for https://github.com/conan-io/conan/issues/12642
-                gnu_triplet = get_gnu_triplet(str(self._settings_build.os), str(self._settings_build.arch), str(self.settings.compiler))
-                tc.configure_args.append(f"--build={gnu_triplet}")
-            if self.settings.os in ["iOS", "tvOS", "watchOS"]:
-                gnu_triplet = get_gnu_triplet("Macos", str(self.settings.arch))
-                tc.configure_args.append(f"--host={gnu_triplet}")
-            elif is_msvc(self):
-                # ICU doesn't like GNU triplet of conan for msvc (see https://github.com/conan-io/conan/issues/12546)
-                host = get_gnu_triplet(str(self.settings.os), str(self.settings.arch), "gcc")
-                build = get_gnu_triplet(str(self._settings_build.os), str(self._settings_build.arch), "gcc")
-                tc.configure_args.extend([
-                    f"--host={host}",
-                    f"--build={build}",
-                ])
```

Comment:
It will break cross-build to iOS/tvOS/watchOS and cross-build with Visual Studio.

Reply:
isn't iOS already broken? #15381

Reply:
I can build icu for iOS armv8 from macOS Intel with latest RREV in conan-center, and conan 1.58.0

Reply:
I confirm that this PR will break cross-build from macOS (Intel for me) to iOS:

```
=== Important Notes: ===
Data Packaging: archive
 This means: ICU data will be stored in a single .dat file.
 To locate data: ICU will look in //lib/icu/72.1 which is the installation location. Call u_setDataDirectory() or use the ICU_DATA environment variable to override.
Building ICU: Use a GNU make such as /usr/bin/gnumake to build ICU.
checking the version of "/usr/bin/gnumake"... 3.81 (we wanted at least 3.80)
ok
C++ apps may want to build with CXXFLAGS =  -std=c++11
## Note: you have disabled ICU's tools. This ICU cannot build its own data or tests.
## Expect build failures in the 'data', 'test', and other directories.
** Note: Python not found. You will not be able to build data from source or run tests.
cd ./config; \
		/Applications/Xcode.app/Contents/Developer/usr/bin/make -f pkgdataMakefile
rm -rf config/icu-config
rebuilding config/icucross.mk
rebuilding config/icucross.inc
/bin/sh /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/mkinstalldirs lib
/bin/sh /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/mkinstalldirs bin
/usr/bin/install -c /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/config/icu-config-top config/icu-config
chmod u+w config/icu-config

mkdir bin
LC_ALL=C /usr/bin/sed -f /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/config/make2sh.sed < ./config/Makefile.inc | grep -v '#M#' | uniq >> config/icu-config
mkdir lib

*** ERROR - configure could not detect your platform
*** see the readme.html
/Applications/Xcode.app/Contents/Developer/usr/bin/make[0]: Making `all' in `stubdata'
*** or, try copying icu/source/config/mh-linux to mh-unknown
config/icu-io.pc updated.
LC_ALL=C /usr/bin/sed -f /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/config/make2sh.sed < /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/config/mh-unknown | grep -v '#M#' | uniq >> config/icu-config
config/icu-i18n.pc updated.
config/icu-uc.pc updated.
***   and editing it.


exit 1

cat /Users/spaceim/.conan/data/icu/72.1/_/_/build/6740f66694763d92a39a7d904b992bb6c3c1ba70/src/source/config/icu-config-bottom >> config/icu-config

rm -rf pkgdata.inc
*** ERROR - configure could not detect your platform
chmod u-w config/icu-config
*** see the readme.html
*** or, try copying icu/source/config/mh-linux to mh-unknown
***   and editing it.


exit 1
make: *** [all-recursive] Error 2
make: *** Waiting for unfinished jobs....
```

Reply:
As inconvenient as it is the build service does not support those scenarios so we can't not enforce they work which means we dont expect contributors to not break them.

Given that they work for some and not others ... it's already hard to say there's a robust solution

Reply:
Seriously? Do I really need to argue about that? You deliberately decide to accept a PR breaking a configuration? Cross-build from macOS Intel to iOS arm works currently, but not from macOS M1 to iOS arm. With this PR, none of them will work.

Reply:
it shouldn't be removed. definitely, not in this PR, as it's goal was advertised as `compatibility tweaks`, but it actually removes some existing compatibility tweaks :exploding_head: